index/scorch/persister.go

```diff
@@ -873,6 +887,17 @@ func (s *Scorch) loadSegment(segmentBucket *bolt.Bucket) (*SegmentSnapshot, erro
 			rv.deleted = deletedBitmap
 		}
 	}
+	statBytes := segmentBucket.Get(boltStatsKey)
+	if statBytes != nil {
+		var stats map[string]map[string]int
```

Comment:
Using a struct would be a better idea keeping future in mind.

Reply:
Changed to use a struct that implements an interface defined in scorch_segment_api. Zap layer uses the functions exported by this interface to add its stats.